lib/openstack/openstack_event_monitor.rb

```diff
@@ -55,6 +58,11 @@ def each_batch
     raise NotImplementedError, "must be implemented in subclass"
   end
 
+  def retrieve_port(options, default_port)
+    port = options[:port].to_i # 0 if options[:port] is nil
+    port = default_port unless port > 0
```

Comment:
I kind of like the way @kbrock did it with `options[:port].presence.try(:to_i)`  Then maybe this method goes away?